File: ECommerce.Domain/Entities/Product.cs
namespace ECommerce.Domain.Entities;

public class Product
{
    public int Id { get; set; }

    // Unique business identifier used for Excel bulk updates.
    public string SKU { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public decimal Price { get; set; }

    public int CategoryId { get; set; }

    public Category Category { get; set; } = null!;

    public int StockQuantity { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}

File: ECommerce.Frontend/src/app/models/product.ts
export interface Product {
  id: number;
  sku: string;
  name: string;
  description: string;
  price: number;
  categoryId: number;
  stockQuantity: number;
}

export interface CreateProduct {
  sku: string;
  name: string;
  description: string;
  price: number;
  categoryId: number;
  stockQuantity: number;
}

export interface UpdateProduct {
  sku: string;
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
  categoryId: number;
}